packages/block-library/src/buttons/style.scss

```diff
@@ -1,37 +1,69 @@
-// Increased specificity to override blocks default margin.
-.wp-block-buttons .wp-block-button {
-	display: inline-block;
-	margin-right: 0.5em;
-	margin-bottom: 0.5em;
+.wp-block-buttons {
+	display: flex;
+	flex-direction: row;
+	flex-wrap: wrap;
 
-	&:last-child {
-		margin-right: 0;
+	// Increased specificity to override blocks default margin.
+	> .wp-block-button {
+		display: inline-block;
+		/*rtl:ignore*/
+		margin-right: 0.5em;
+		margin-bottom: 0.5em;
+
+		&:last-child {
+			/*rtl:ignore*/
+			margin-right: 0;
+		}
 	}
-}
 
-.wp-block-buttons.alignright .wp-block-button {
-	/*rtl:ignore*/
-	margin-right: 0;
-	/*rtl:ignore*/
-	margin-left: 0.5em;
+	&.is-content-justification-left {
+		justify-content: flex-start;
```

Comment:
At the same time as preferring the word 'items' in my earlier comment, I do like how tidy the classname and css is 😄